tools/pythonpkg/tests/fast/test_filesystem.py

```diff
@@ -184,3 +184,24 @@ def test_database_attach(self, tmp_path: Path, monkeypatch: MonkeyPatch):
         # duckdb sometimes seems to swallow write errors, so we use this to ensure that 
         # isn't happening
         assert not write_errors
+
+    def test_copy_partition(self, duckdb_cursor: DuckDBPyConnection, memory: AbstractFileSystem):
+        duckdb_cursor.register_filesystem(memory)
+
+        duckdb_cursor.execute("copy (select 1 as a) to 'memory://root' (partition_by (a))")
+
+        assert memory.open(
+            '/root\\a=1\\data_0.csv'
```

Comment:
This is to work around the fact that DuckDB defaults to using the path separator of the system it's running on, rather than the path separator of the filesystem